diag_manager/fms_diag_buffer.F90

```diff
@@ -0,0 +1,1546 @@
+  type is(integer(i4_kind))
+    select type(fillval)
+    type is(integer(i4_kind))
+      buff = fillval
+    class default
+      call mpp_error(FATAL, 'initialize_buffer_4d: fillval does not match up with allocated buffer type(i4_kind)' // &
+                            ' for field' // field_name_str )
+    end select
+  class default
+    call mpp_error(FATAL, 'initialize buffer_4d: buffer allocated to invalid data type, this shouldnt happen')
+  end select
+
+end subroutine initialize_buffer_4d
+
+!> @brief Initializes a buffer to a given fill value
+subroutine initialize_buffer_5d (this, fillval, field_name)
+  class(buffer5d_type), intent(inout) :: this !< allocated 5D buffer object
+  class(*), intent(in) :: fillval!< fill value, must be same type as the allocated buffer in this
+  character(len=*), optional, intent(in) :: field_name !< optional field name for error output
+  character(len=128)          :: field_name_str
+  if(present(field_name)) then
+    field_name_str = field_name(1:128)
+  else
+    field_name_str = 'N/A'
+  endif
+
+  if(.not. allocated(this%buffer)) call mpp_error(FATAL, 'initialize_buffer_5d: field:'// field_name_str // &
+      'buffer not yet allocated, allocate_buffer() must be called on this object first.')
+  ! have to check fill value and buffer types match
+  select type(buff => this%buffer)
+  type is(real(r8_kind))
+    select type(fillval)
+    type is(real(r8_kind))
+      buff = fillval
+    class default
+      call mpp_error(FATAL, 'initialize_buffer_5d: fillval does not match up with allocated buffer type(r8_kind)' // &
+                            ' for field' // field_name_str )
+    end select
+  type is(real(r4_kind))
+    select type(fillval)
+    type is(real(r4_kind))
+      buff = fillval
+    class default
+      call mpp_error(FATAL, 'initialize_buffer_5d: fillval does not match up with allocated buffer type(r4_kind)' // &
+                            ' for field' // field_name_str )
+    end select
+  type is(integer(i8_kind))
+    select type(fillval)
+    type is(integer(i8_kind))
+      buff = fillval
+    class default
+      call mpp_error(FATAL, 'initialize_buffer_5d: fillval does not match up with allocated buffer type(i8_kind)' // &
+                            ' for field' // field_name_str )
+    end select
+  type is(integer(i4_kind))
+    select type(fillval)
+    type is(integer(i4_kind))
+      buff = fillval
+    class default
+      call mpp_error(FATAL, 'initialize_buffer_5d: fillval does not match up with allocated buffer type(i4_kind)' // &
+                            ' for field' // field_name_str )
+    end select
+  class default
+    call mpp_error(FATAL, 'initialize buffer_5d: buffer allocated to invalid data type, this shouldnt happen')
+  end select
+
+end subroutine initialize_buffer_5d
+
+!> @brief Add values to 0d buffer
+!! this will just call the init routine since there's only one value
+!! input_data must match allocated type of buffer object
```

Comment:
Will this print out as one line in the doxygen documentation page?  I think it needs some capitalization and punctuation

Reply:
Yeah it will I added some capitalization and punctuation to all the routines